consensus/utils.go

```diff
@@ -0,0 +1,57 @@
+package consensus
+
+import (
+	"reflect"
+
+	"github.com/deso-protocol/core/collections"
+)
+
+func isValidBlock(block Block) bool {
```

Comment:
Okay, so in addition to this validation check, we will also have a broader sanity check that verifies the signature, the QC, etc. This would correspond to the `Node.sanityCheckBlock` in [Hotstuff reference](https://github.com/deso-protocol/hotstuff_pseudocode/pull/7).

It's not a necessity, but I was just thinking about whether `isValidBlock` is the right name, given that it's really the other function that checks for the block's validity. I don't have a better alternative, but maybe just think about it.

Reply:
Yup! I'm aligned. This check performs the subset of checks from the reference that it has the knowledge to perform:
- Ensure the block height & view aren't malformed
- Ensure the QC isn't malformed
It performs these checks so that it's impossible for the code to panic here due to a null pointer check.

It expects the higher level checks to have been performed in blockchain.go by the caller
- Ensure the block's view isn't stale
- Ensure the block was proposed by the expected leader
- Ensure the QC has a valid supermajority vote
- Validate all signatures and txns

You're right that `isValidBlock` is probably not the clearest name. Maybe `isProperlyFormedBlock`. I'll update the name